src/main.c

```diff
@@ -0,0 +1,288 @@
+	char *boot_order;
+
+	if (args->boot_order == NULL)
+		return true;
+
+	boot_order = strdup(args->boot_order);
+
+	for (ptr = boot_order; (token = strtok(ptr, ",")) != NULL; ptr = NULL) {
+		int i;
+		for (i = 0; i < boot->record_count; ++i) {
+			if (strcmp(boot->records[i].name, token) == 0) {
+				boot_data_move(boot, i, target++);
+				break;
+			}
+		}
+
+		if (i == boot->record_count) {
+			fprintf(stderr, "Unrecognized boot record name: %s\n",
+				token);
+			break;
+		}
+	}
+
+	free(boot_order);
+
+	return (token == NULL);
+}
+
+static bool set_option(struct option *option, const char *str_value)
+{
+	int int_value = 0;
+	const struct option_def *option_def = &OPTIONS[option->id];
+
+	switch (option_def->type) {
+		case OPT_TYPE_BOOLEAN:
+			if (strcmp(str_value, "on") == 0)
+				int_value = 1;
+			else if (strcmp(str_value, "off") == 0)
+				int_value = 0;
+			else
+				return false;
+			break;
+		case OPT_TYPE_TOGGLE:
+			if (strcmp(str_value, "first") == 0)
+				int_value = 0;
+			else if (strcmp(str_value, "second") == 0)
+				int_value = 1;
+			else
+				return false;
+			break;
+		case OPT_TYPE_HEX4:
+			int_value = strtol(str_value, NULL, 10);
+			break;
+	}
+
+	return boot_data_set_option(option, int_value);
+}
+
+static bool batch_set_options(const struct args *args, struct boot_data *boot)
+{
+	int i;
+
+	for (i = 0; i < args->boot_option_count; ++i) {
+		int n;
+		int j;
+
+		char name[64];
+		char value[64];
+
+		n = sscanf(args->boot_options[i], "%63[^=]=%63s", name, value);
+		if (n != 2) {
+			fprintf(stderr, "Unrecognized option setting: %s\n",
+				args->boot_options[i]);
+			break;
+		}
+
+		for (j = 0; j < boot->option_count; ++j) {
+			const int id = boot->options[j].id;
+			const struct option_def *option_def = &OPTIONS[id];
+			if (strcmp(option_def->keyword, name) == 0) {
+				break;
+			}
+		}
+
+		if (j == boot->option_count) {
+			fprintf(stderr, "Unrecognized option: %s\n", name);
+			break;
+		}
+
+		if (!set_option(&boot->options[j], value)) {
+			fprintf(stderr, "Invalid value for %s option: %s\n",
+				name, value);
+			break;
+		}
+
+	}
+
+	return (i == args->boot_option_count);
+}
+
+static bool run_batch(const struct args *args, struct boot_data *boot)
+{
+	return batch_reorder(args, boot) &&
+	       batch_set_options(args, boot) &&
+	       cbfs_store_boot_data(args->cbfs_tool, boot, args->rom_file);
+}
+
+static void print_help(const char *command)
+{
+	size_t i;
+
+	printf(USAGE_FMT, command);
+
+	printf("\n");
+	printf("boot-source is a value from a boot order list.\n");
+	printf("\n");
+	printf("Recognized options and possible values:\n");
+
+	for (i = 0; i < ARRAY_SIZE(OPTIONS); ++i) {
+		const struct option_def *option_def = &OPTIONS[i];
+		printf("  %s\n", option_def->keyword);
+		printf("    %s\n", option_def->description);
+
+		switch (option_def->type) {
+			case OPT_TYPE_BOOLEAN:
+				printf("    on/off\n");
+				break;
+			case OPT_TYPE_TOGGLE:
+				printf("    first/second\n");
```

Comment:
What does `second` refer to? Boot option priority?

Reply:
No. "first"/"second" correspond to options like "UART D / GPIO[10..17]" and pick first or second function.